indexer/services/vulcan/src/lib/on-batch.ts

```diff
@@ -47,29 +46,10 @@ export async function onBatch(
     metricTags,
   );
 
-  let lastCommitTime: number = startTime;
   for (let i = 0; i < batch.messages.length; i++) {
     const message: KafkaMessage = batch.messages[i];
     await onMessage(message);
-
-    // Commit every KAFKA_BATCH_PROCESSING_COMMIT_FREQUENCY_MS to reduce number of roundtrips, and
-    // also prevent disconnecting from the broker due to inactivity.
-    const now: number = Date.now();
-    if (now - lastCommitTime > config.KAFKA_BATCH_PROCESSING_COMMIT_FREQUENCY_MS) {
-      logger.info({
-        at: 'on-batch#onBatch',
-        message: 'Committing offsets and sending heart beat',
-        ...batchInfo,
-      });
-      payload.resolveOffset(message.offset);
-      await Promise.all([
-        payload.heartbeat(),
```

Comment:
are you sure that `heartbeatInterval` being set means that it's automatically heartbeating without calling `payload.heartbeat()`? From the KafkaJS docs I read it as just gating the frequency at which `payload.heartbeat()` actually sends a heartbeat

Reply:
Based on docs https://kafka.js.org/docs/consuming#a-name-options-a-options, I was under assumption that heartbeat is sent every `heartbeatInterval`. Not explicitly sending heartbeats works in internal mainnet, so I'm assuming specifying `heartbeatInterval` must actually send a heartbeat, otherwise we'd reach sessionTimeout & be disconnected.